### PR TITLE
Add utf8casestr (case insensitive)

### PR DESCRIPTION
**Pull request by MattSturgeon**
@Chloridite added this function for use in [OpenBoardView](/OpenBoardView/OpenBoardView). @piernov asked me to extract it into a commit, so I figured I may as well send a PR while I was at it :smiley: 

`utf8casestr` is a case *in*sensitive version of the existing `utf8str`

`void* utf8casestr (const void* haystack, const void* needle)`

### CODE REVIEW

**Comment by sheredom** [comment]
@MattSturgeon thanks for this! I'll have a look and see what the build failures are and get it in :smile:

File: utf8.h [hunk 1]
```diff
@@ -662,6 +665,54 @@ void* utf8str(const void* haystack, const void* needle) {
...
+	// byte then return haystack
+	if ('\0' == *((const char*)needle)) {
+		return (void*)haystack;
+	}
+
+	while ('\0' != *h) {
+		const char* maybeMatch = h;
+		const char* n = (const char*)needle;
+
+		while (true) {
```
Comment (MattSturgeon) [inline utf8.h]: Looks like this is causing the build error. OpenBoardView is C++, hence why I didn't notice

**Comment by MattSturgeon** [comment]
Should build ok now.

**Comment by MattSturgeon** [comment]
Replaced by #18.

**Pull request closed without merging**